src/jdk.httpserver/share/classes/sun/net/httpserver/simpleserver/SimpleFileServerImpl.java

```diff
@@ -112,7 +112,7 @@ static int start(PrintWriter writer, String launcher, String[] args) {
                         addrSpecified = true;
                     }
                     case "-d", "--directory" ->
-                        root = Path.of(optionArg = options.next());
+                        root = Path.of(optionArg = options.next()).normalize().toAbsolutePath();
```

Comment:
normalize removes redundant name elements and is for "user input". Are you 100% sure it make sense here?

Reply:
@AlanBateman, thanks for the review. In 5f9e3df8f607d82987ca748492e6b0468462cb14, replaced `normalize().toAbsolutePath()` with `toRealPath()`.